### PR TITLE
Kmt enhancements

### PR DESCRIPTION
**Pull request by dgem**
This pull requests covers most of our known requirements for changes in myFT to support our usecases (kmt-usercases.spec.js).

It adds in two new models, Group and License. A License can have many Users and Groups as members of the license. Using the existing API it allows us to manage users and groups on a license.

The createRelationship API has been enhanced so that multiple things can have relationships to many things. This is so we can batch add a large number of users to a license, for example.

It also adds a somewhat experimental scopedReadAll. Specifically it's been added and tested so that we can ask "Which users are following this topic on for this license". This might need some refinement or further work to make it useful for myFT as a whole. To mitigate against use the URL is filtered so that only the specific question can be asked...

Questions, comments and suggestions welcome.

### CODE REVIEW

File: test/v3/kmt-usecases.spec.js [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+			}
+			request(api)
+			.post(`${licensePath}${licenseId}/member/${type}`)
+			.send(data)
+			.end ((err, res) => {
+				logResponse(err,res);
+				expect(res.status).to.equal(200);
+				if (Array.isArray(ids)) {
+					expect(res.body.length).to.equal(ids.length);
+          ids.every(id=>{
```
Comment (laurieboyes) [inline test/v3/kmt-usecases.spec.js]: \*Puts on pedant hat 🎩 \* - some of this indenting seems to have gone a bit wonky (or is it just github?)

**Comment by laurieboyes** [comment]
👍  Looks good to me!

File: test/v3/kmt-usecases.spec.js (hunk 1 above)
Comment (dgem) [inline test/v3/kmt-usecases.spec.js]: hmm, doesn't look so nice on github does it... probably a soft versus hard tab thing...

Comment (laurieboyes) [inline test/v3/kmt-usecases.spec.js]: Cool, good enough for me

**Comment by dgem** [comment]
Think this looks better now, was it just this file ?

**Comment by laurieboyes** [comment]
Think I spotted a couple more dotted around but nothing worth delaying merging over 🚢

**Comment by dgem** [comment]
Hi Chaps,
I missed the 🚢 comment and as per slack msg, it seems a little late on Friday pm to merge. Could I ask a favour and that you merge it in on Monday once any weekend fun & games has been dealt with ?
Thanks

**Pull request merged**